packages/components/src/radio-group/radio.tsx

```diff
@@ -0,0 +1,45 @@
+/**
+ * WordPress dependencies
+ */
+import { forwardRef, useContext } from '@wordpress/element';
+
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import * as Ariakit from '@ariakit/react';
+
+/**
+ * Internal dependencies
+ */
+import Button from '../button';
+import { RadioGroupContext } from './context';
+import type { WordPressComponentProps } from '../context';
+import type { RadioProps } from './types';
+
+function UnforwardedRadio(
+	{ value, children }: WordPressComponentProps< RadioProps, 'button', false >,
+	ref: React.ForwardedRef< any >
+) {
+	const { store, disabled } = useContext( RadioGroupContext );
+
+	const selectedValue = store?.useState( 'value' );
+	const isChecked = selectedValue !== undefined && selectedValue === value;
+
+	return (
+		<Ariakit.Radio
+			disabled={ disabled }
+			store={ store }
+			ref={ ref }
+			value={ value }
+			render={
+				<Button variant={ isChecked ? 'primary' : 'secondary' } />
+			}
+		>
+			{ children || value }
+		</Ariakit.Radio>
+	);
+}
+
+export const Radio = forwardRef( UnforwardedRadio );
```

Comment:
We should add JSDoc comments here, like in the older version of the component (specifically about the fact that this component is deprecated)

Reply:
Fixed via https://github.com/WordPress/gutenberg/pull/55580/commits/d31df386d0490ecc27899f3879c3314cd6f6b5c0